src/Orleans.Runtime/Hosting/DefaultSiloServices.cs

```diff
@@ -169,7 +169,9 @@ internal static void AddDefaultServices(HostBuilderContext context, IServiceColl
             services.TryAddSingleton<VersionSelectorManager>();
             services.TryAddSingleton<CachedVersionSelectorManager>();
             // Version selector strategy
-            services.TryAddSingleton<IVersionStore, GrainVersionStore>();
+            services.TryAddSingleton<GrainVersionStore>();
+            services.AddFromExisting<IVersionStore, GrainVersionStore>();
```

Comment:
Suggest
services.TryAddSingleton<IVersionStore>(sp => sp.GetRequiredService<GrainVersionStore>());
As this would allow users to provide their own impl of IVersionStore prior to registering the default services.